src/app/[communitySlug]/topup/[amount]/route.js

```diff
@@ -56,11 +57,32 @@ export async function GET(request, { params }) {
     console.log(">>> topup", row);
 
     try {
-      row.txHash = await transfer(
-        communitySlug,
-        row.amount,
-        row.accountAddress
+      const config = await getConfig(communitySlug);
+      if (!config) {
+        throw new Error(`Community not found (${communitySlug})`);
+      }
+
+      const provider = new ethers.providers.JsonRpcProvider({
+        url: config.node.url,
+        skipFetchSetup: true,
+      });
+
+      const faucetWallet = new Wallet(process.env.FAUCET_PRIVATE_KEY);
+      const signer = faucetWallet.connect(provider);
+
+      const signature = await userOpERC20Transfer(
+        config,
+        provider,
+        signer,
+        accountAddress,
+        amount
       );
+
+      row.txHash = signature;
+
+      // wait 2 seconds
+      // TODO: remove when we have a better way to wait for the tx to be mined
+      await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
Let's remove that. Most of the time, people will open it from the app, so it can directly close the top up window.

It's better to move that logic to the frontend to poll for the new balance.